database/user.go

```diff
@@ -631,3 +677,22 @@ func monthStart(subscribedUntil time.Time) time.Time {
 	d := now.AddDate(0, monthsDelta, daysDelta)
 	return time.Date(d.Year(), d.Month(), d.Day(), 0, 0, 0, 0, time.UTC)
 }
+
+// generateSub is a helper method that generates a UUID and encodes it in hex.
+func generateSub() string {
+	uid, err := uuid.New()
+	// the only way to get an error here is for uuid to be unable to read from
+	// the RNG reader, which is highly unlikely. We'll log and retry.
```

Comment:
Highly unlikely is not impossible :-)
What is the harm in returning the error?
We should at least `build.Critical` and return and empty string instead of just logging the error. That is more in line with how we handle similar cases.

Reply:
Hm. Initially I was very reluctant to use `build.Critical` because that's something we use in order to keep an eye on systems we control. `crypto/rand` is definitely not such a system. OTOH, if we ever fail to read from `crypto/rand`'s reader that's probably a non-recoverable error that we might want to `log.Fatal` on, as our node will be unable to create new IDs.

In any case, I'll add the `build.Critical`, although it seems like a bad fit for the case because it's not in a module we control.